src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -579,20 +590,18 @@ void Chttp2ServerListener::ActiveConnection::SendGoAway() {
     MutexLock lock(&mu_);
     if (transport_ != nullptr && !shutdown_) {
       transport = transport_;
-      Ref().release();  // Ref held by OnDrainGraceTimeExpiry
-      GRPC_CLOSURE_INIT(&on_drain_grace_time_expiry_, OnDrainGraceTimeExpiry,
-                        this, nullptr);
-      grpc_timer_init(
-          &drain_grace_timer_,
-          Timestamp::Now() +
-              std::max(
-                  Duration::Zero(),
-                  listener_->args_
-                      .GetDurationFromIntMillis(
-                          GRPC_ARG_SERVER_CONFIG_CHANGE_DRAIN_GRACE_TIME_MS)
-                      .value_or(Duration::Minutes(10))),
-          &on_drain_grace_time_expiry_);
-      drain_grace_timer_expiry_callback_pending_ = true;
+      drain_grace_timer_handle_ = event_engine_->RunAfter(
+          std::max(Duration::Zero(),
+                   listener_->args_
+                       .GetDurationFromIntMillis(
+                           GRPC_ARG_SERVER_CONFIG_CHANGE_DRAIN_GRACE_TIME_MS)
+                       .value_or(Duration::Minutes(10))),
+          [self = Ref(DEBUG_LOCATION, "drain_grace_timer")]() mutable {
+            ApplicationCallbackExecCtx callback_exec_ctx;
```

Comment:
question for @drfloob : Do all timers now need ExecCtx to be declared on the stack?

Reply:
_Need_ is a string word, but yes, the policy is that all callbacks must create both ExecCtx types locally at the moment. Debugging missing ExecCtxs in EventEngine callbacks has been complicated, and it will be trivially easy to remove them all once we've migrate away from `ExecCtx::Run` (a long-term thing).